src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(state, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          if (state[key]) {
+            delete state[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        Object.keys(state).forEach(prop => delete state[prop]);
+        break;
```

Comment:
There's actually no need in this break, cause you don't have conditions after this one